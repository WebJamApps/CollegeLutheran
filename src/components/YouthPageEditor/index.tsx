import React from 'react';
import type { AdminDashboard } from '../../containers/AdminDashboard';

type PageProps = {comp:AdminDashboard;youthTitle:string;youthContent:string};
const YouthPageEditor = ({ comp, youthTitle, youthContent }:PageProps): JSX.Element => (
  <div className="horiz-scroll">
    <div className="material-content elevation3" style={{ width: '850px', margin: '30px auto' }}>
      <h5>Change Youthpage Section</h5>
      <form
        id="update-youthpage"
        style={{
          textAlign: 'left', marginLeft: '4px', width: '100%', maxWidth: '100%',
        }}
      >
        {comp.forms.makeInput({
          type: 'text',
          label: 'Youth Title',
          isRequired: false,
          onChange: (evt) => comp.setState({ youthTitle: evt.target.value }),
          value: youthTitle,
          width: '90%',
        })}
        <label htmlFor="content">
          Content
          <br />
          {comp.controller.editor(youthContent, comp.controller.onChangeYouthContent)}
        </label>
        <div style={{ marginLeft: '60%', marginTop: '10px' }}>
          <button
            type="button"
            id="update-youthContent"
            disabled={false}
            onClick={(evt) => comp.controller.putAPI(evt, { title: youthTitle, comments: youthContent, type: 'youthPageContent' }, '/youth')}
          >
            Update Youthpage
          </button>
        </div>
      </form>
    </div>
  </div>
);

export default YouthPageEditor;
