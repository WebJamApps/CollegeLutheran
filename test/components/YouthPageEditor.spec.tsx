/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import YouthPageEditor from '../../src/components/YouthPageEditor';
import Forms from '../../src/lib/forms';

describe('YouthPageEditor', () => {
  let wrapper:ShallowWrapper, adminDashboardStub:any = {};
  beforeEach(() => {
    adminDashboardStub = {
      setState: jest.fn(),
      onChangeAdminEmail: jest.fn(),
      state: { formError: 'bad' },
      controller: { validateAdmin: jest.fn(), editor: jest.fn(), putAPI: jest.fn() },
      forms: { makeInput: () => ({ onChange: () => adminDashboardStub.setState() }) },
    };
    wrapper = shallow(<YouthPageEditor comp={adminDashboardStub} youthTitle="Youth News" youthContent="Howdy!" makeInput={Forms.makeInput} />);
  });
  it('renders correctly', () => { expect(wrapper).toMatchSnapshot(); });
  it('handles onChange youthTitle', () => {
    wrapper.find('input').simulate('change', { target: { value: 'Youth News' } });
    expect(adminDashboardStub.setState).toHaveBeenCalled();
  });
  it('click button to submit', () => {
    wrapper.find('button').simulate('click', { preventDefault: jest.fn() });
    expect(adminDashboardStub.controller.putAPI).toHaveBeenCalled();
  });
});
