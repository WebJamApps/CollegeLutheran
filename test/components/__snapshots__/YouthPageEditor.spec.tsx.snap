// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouthPageEditor renders correctly 1`] = `ShallowWrapper {}`;
